# Kitchen P&L Data Analysis

Analyst : Rohit Kumar

Date : May 2026

Objective : Analyze cloud-kitchen P&L data and build two dashboards (Kitchen-level P&L and Variance-level P&L)

---

## Phase 1 : Data Interrogation
Before any analysis, we need to understand what are working with.
We look upon these 8 questions about the dataset before writing any insight.

#### The 9 Questions
1. How big is the dataset ?
2. What columns exist and what are there types?
3. What are the data types ?
4. What do the first and last row look like ?
5. Are there any missing values ?
6. Are there any duplicates?
7. What are the unique value in each category column?
8. What is the range of numerical values?
9. What is thr time range covered ?


In [2]:
import pandas as pd
import numpy as np
import warnings
import sys

In [3]:
# Display 
pd.set_option('display.max_columns',None)
pd.set_option('display.float_format','{;,.2f}'.format)



In [4]:
warnings.filterwarnings('ignore')

In [5]:
print("=== Environment Details ===")
print(f"Python Version : {sys.version.split()[0]}")
print(f"Pandas Version : {pd.__version__}")
print(f"NumPy Version  : {np.__version__}")

=== Environment Details ===
Python Version : 3.12.7
Pandas Version : 3.0.3
NumPy Version  : 2.4.4


In [6]:
## Load Dataset
file_path = "../data/kitchen_pnl.xlsx"
df = pd.read_excel(file_path, header=1)


In [7]:
# === Dataset Size ===
rows, cols = df.shape
memory_mb = df.memory_usage(deep=True).sum()/1024**2
print(f"Rows : {rows:,}")
print(f"Columns : {cols}")
print(f"Memory size : {memory_mb:.2f} MB")

Rows : 2,100
Columns : 17
Memory size : 0.46 MB


In [8]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 2100 entries, 0 to 2099
Data columns (total 17 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   MONTH            2100 non-null   str    
 1   CITY             2100 non-null   str    
 2   STORE            2100 non-null   str    
 3   STATUS           2100 non-null   str    
 4   ZONE MAPPING     2100 non-null   str    
 5   ORDER COUNT      2100 non-null   int64  
 6   CART SALES       2100 non-null   float64
 7   DISCOUNT         2100 non-null   float64
 8   NET REVENUE      2100 non-null   float64
 9   IDEAL FOOD COST  2100 non-null   float64
 10  GROSS MARGIN     2100 non-null   float64
 11  KITCHEN EBITDA   2100 non-null   float64
 12  VARIANCE         2100 non-null   float64
 13  REVENUE COHORT   2100 non-null   str    
 14  CM COHORT        2100 non-null   str    
 15  EBITDA CATEGORY  2100 non-null   str    
 16  EBITDA COHORT    2100 non-null   str    
dtypes: float64(7), int64(1), 

In [9]:
# Reset display settings
pd.reset_option('display.max_columns')
pd.reset_option('display.float_format')

In [10]:
# === Sample Data Preview ===
print("First 5 Rows")
df.head(5)


First 5 Rows


,MONTH,CITY,STORE,STATUS,ZONE MAPPING,ORDER COUNT,CART SALES,DISCOUNT,NET REVENUE,IDEAL FOOD COST,GROSS MARGIN,KITCHEN EBITDA,VARIANCE,REVENUE COHORT,CM COHORT,EBITDA CATEGORY,EBITDA COHORT
0,Oct-2023,Ahmedabad,Ramos-Martin,Inactive,West,14270,6554333.68,1163493.98,5434068.17,1763679.62,3614747.39,2374914.55,29664.62,INR 20 to 30 lacs,0% to 10%,EBITDA +ve,More than 20%
1,Nov-2023,Ahmedabad,Ramos-Martin,Inactive,West,8528,2637625.82,636015.27,2146679.55,638415.30,1414249.00,151761.82,11769.85,INR 20 to 30 lacs,More than 30%,EBITDA +ve,More than 20%
2,Dec-2023,Ahmedabad,Ramos-Martin,Inactive,West,10798,4364193.67,1283840.18,3188606.71,1085080.58,2036999.44,736811.76,23376.83,INR 30 to 40 lacs,0% to 10%,EBITDA +ve,0% to 10%
3,Jan-2024,Ahmedabad,Ramos-Martin,Inactive,West,7804,3479969.32,854797.46,2734306.33,1180892.50,1451140.55,229924.26,26162.41,INR 20 to 30 lacs,0% to 10%,EBITDA +ve,More than 20%
4,Feb-2024,Ahmedabad,Ramos-Martin,Inactive,West,10060,4771378.71,1290900.38,3602124.88,1114846.94,2394567.16,1114151.65,19944.97,INR 20 to 30 lacs,More than 30%,EBITDA -ve,0% to 10%


In [11]:
print("Last 5 Rows")
df.tail()

Last 5 Rows


,MONTH,CITY,STORE,STATUS,ZONE MAPPING,ORDER COUNT,CART SALES,DISCOUNT,NET REVENUE,IDEAL FOOD COST,GROSS MARGIN,KITCHEN EBITDA,VARIANCE,REVENUE COHORT,CM COHORT,EBITDA CATEGORY,EBITDA COHORT
2095,Nov-2023,Mumbai,"Jones, Knight and Rivers",Active,East,12005,3764132.03,650583.53,3236050.77,1368304.24,1749922.82,546305.61,24124.15,More than 40 lacs,0% to 10%,EBITDA +ve,More than 20%
2096,Dec-2023,Mumbai,"Jones, Knight and Rivers",Active,East,11454,4776944.70,763880.67,4078760.77,1795668.85,2208469.29,939926.29,28627.88,INR 30 to 40 lacs,0% to 10%,EBITDA +ve,More than 20%
2097,Jan-2024,Mumbai,"Jones, Knight and Rivers",Active,East,12568,5683969.21,1386463.66,4378116.35,1815618.84,2486078.29,1375840.66,11235.39,INR 20 to 30 lacs,10% to 20%,EBITDA -ve,10% to 20%
2098,Feb-2024,Mumbai,"Jones, Knight and Rivers",Active,East,12229,5031251.50,797713.07,4335963.51,2023942.25,2208644.82,955765.94,26345.33,More than 40 lacs,0% to 10%,EBITDA +ve,More than 20%
2099,Mar-2024,Mumbai,"Jones, Knight and Rivers",Active,East,7955,2422156.10,597663.92,1921713.45,871664.51,926452.93,-360347.81,11966.10,INR 20 to 30 lacs,More than 30%,EBITDA -ve,More than 20%


In [12]:
#=== Duplicate Check ===
print("=== Duplicate Check ===\n")

exact_dups = df.duplicated().sum()
key_dups = df.duplicated(subset=['MONTH', 'STORE']).sum()

print(f"Exact duplicate rows           : {exact_dups}")
print(f"Duplicate (MONTH, STORE) pairs : {key_dups}")

=== Duplicate Check ===

Exact duplicate rows           : 0
Duplicate (MONTH, STORE) pairs : 36


In [13]:
duplicate_store_month = df[
    df.duplicated(subset=['MONTH', 'STORE'], keep=False)
]

duplicate_store_month.sort_values(
    ['STORE', 'MONTH']
).head(10)

,MONTH,CITY,STORE,STATUS,ZONE MAPPING,ORDER COUNT,CART SALES,DISCOUNT,NET REVENUE,IDEAL FOOD COST,GROSS MARGIN,KITCHEN EBITDA,VARIANCE,REVENUE COHORT,CM COHORT,EBITDA CATEGORY,EBITDA COHORT
290,Dec-2023,Ahmedabad,Adams LLC,Active,South,10375,4805006.00,765957.96,4146372.64,1231410.60,2804082.24,1527345.75,26603.85,INR 20 to 30 lacs,More than 30%,EBITDA +ve,0% to 10%
482,Dec-2023,Ahmedabad,Adams LLC,Inactive,East,12388,5336266.42,876427.34,4538510.90,1577613.19,2887443.30,1654361.51,21364.17,INR 20 to 30 lacs,10% to 20%,EBITDA -ve,10% to 20%
292,Feb-2024,Ahmedabad,Adams LLC,Active,South,10508,3417076.85,975454.57,2545862.26,1106893.37,1294872.31,-191137.84,20968.02,INR 20 to 30 lacs,0% to 10%,EBITDA -ve,0% to 10%
484,Feb-2024,Ahmedabad,Adams LLC,Inactive,East,11897,5697872.15,864617.86,4875847.30,1724181.14,3021089.78,1832823.25,11482.38,More than 40 lacs,More than 30%,EBITDA +ve,10% to 20%
291,Jan-2024,Ahmedabad,Adams LLC,Active,South,9727,4276519.50,774758.20,3563156.64,1581067.01,1856259.37,321450.81,14218.55,INR 30 to 40 lacs,0% to 10%,EBITDA +ve,10% to 20%
483,Jan-2024,Ahmedabad,Adams LLC,Inactive,East,7003,3352738.67,632027.00,2781729.09,1314500.33,1391664.56,-28323.76,29553.35,INR 30 to 40 lacs,0% to 10%,EBITDA -ve,0% to 10%
293,Mar-2024,Ahmedabad,Adams LLC,Active,South,9827,4586020.38,927311.87,3790888.22,1098421.91,2572074.97,1103324.10,21742.66,INR 20 to 30 lacs,More than 30%,EBITDA +ve,0% to 10%
485,Mar-2024,Ahmedabad,Adams LLC,Inactive,East,14285,6832627.47,1032481.88,5882855.46,1852343.38,3958408.57,2544886.12,14212.05,More than 40 lacs,10% to 20%,EBITDA +ve,10% to 20%
289,Nov-2023,Ahmedabad,Adams LLC,Active,South,8250,3646785.89,917535.99,2828398.66,1312047.58,1374205.31,-66167.33,17301.88,More than 40 lacs,0% to 10%,EBITDA +ve,More than 20%
481,Nov-2023,Ahmedabad,Adams LLC,Inactive,East,10392,3591659.90,709539.90,2935494.03,1088373.23,1746053.86,431254.10,20990.45,More than 40 lacs,0% to 10%,EBITDA +ve,10% to 20%


In [14]:
key_dups_strict = df.duplicated(subset=['MONTH', 'STORE', 'ZONE MAPPING']).sum()
print(f"Duplicate (MONTH, STORE, ZONE) combinations: {key_dups_strict}")

Duplicate (MONTH, STORE, ZONE) combinations: 6


In [15]:
remaining_dups = df[
    df.duplicated(
        subset=['MONTH', 'STORE', 'ZONE MAPPING'],
          keep=False)]
print(f"Total rows involved: {len(remaining_dups)}")
remaining_dups.sort_values(['STORE', 'MONTH'])

Total rows involved: 12


,MONTH,CITY,STORE,STATUS,ZONE MAPPING,ORDER COUNT,CART SALES,DISCOUNT,NET REVENUE,IDEAL FOOD COST,GROSS MARGIN,KITCHEN EBITDA,VARIANCE,REVENUE COHORT,CM COHORT,EBITDA CATEGORY,EBITDA COHORT
272,Dec-2023,Bangalore,Smith Ltd,Active,West,8340,3619032.87,1064746.14,2618768.50,1198075.22,1317909.88,-174465.72,22769.10,INR 30 to 40 lacs,More than 30%,EBITDA +ve,10% to 20%
1736,Dec-2023,Ahmedabad,Smith Ltd,Active,West,8637,2952244.16,458092.03,2626918.01,872427.12,1637693.01,192514.76,18949.16,INR 20 to 30 lacs,10% to 20%,EBITDA +ve,0% to 10%
274,Feb-2024,Bangalore,Smith Ltd,Active,West,12947,5130088.60,1123106.39,4095073.54,1958205.63,2015341.20,593785.38,18222.98,INR 30 to 40 lacs,More than 30%,EBITDA -ve,0% to 10%
1738,Feb-2024,Ahmedabad,Smith Ltd,Active,West,13399,6317354.99,1431008.42,4977712.07,1567868.13,3325825.20,1914148.10,27796.12,More than 40 lacs,More than 30%,EBITDA -ve,More than 20%
273,Jan-2024,Bangalore,Smith Ltd,Active,West,7430,3632898.62,811525.75,2901868.37,1078949.21,1700206.04,487171.91,11130.31,INR 30 to 40 lacs,More than 30%,EBITDA -ve,More than 20%
1737,Jan-2024,Ahmedabad,Smith Ltd,Active,West,14444,7026178.90,1120035.54,6013015.47,1775318.83,4172729.98,2834680.28,25404.48,INR 20 to 30 lacs,10% to 20%,EBITDA -ve,0% to 10%
275,Mar-2024,Bangalore,Smith Ltd,Active,West,12922,5012463.91,802425.77,4309231.30,1268231.60,2921200.65,1676649.49,17156.82,More than 40 lacs,0% to 10%,EBITDA -ve,0% to 10%
1739,Mar-2024,Ahmedabad,Smith Ltd,Active,West,11917,5397429.17,1202323.81,4280840.46,1320646.79,2826584.68,1536384.53,13633.28,INR 30 to 40 lacs,More than 30%,EBITDA -ve,10% to 20%
271,Nov-2023,Bangalore,Smith Ltd,Active,West,13838,5326447.15,1544945.77,3825607.46,1604873.34,2149868.93,816706.08,14944.12,INR 30 to 40 lacs,0% to 10%,EBITDA -ve,0% to 10%
1735,Nov-2023,Ahmedabad,Smith Ltd,Active,West,11872,4413593.80,675553.34,3791830.69,1481772.80,2238240.87,812506.89,23917.24,More than 40 lacs,More than 30%,EBITDA -ve,10% to 20%


In [16]:
# === Kitchen Coverage Analysis ===
unique_brands = df['STORE'].nunique()
unique_kitchens = df.groupby(['CITY', 'STORE', 'ZONE MAPPING']).ngroups
print(f"Unique store brand names    : {unique_brands}")
print(f"Unique kitchen locations    : {unique_kitchens}")
print(f"Difference (multi-location) : {unique_kitchens - unique_brands}")

Unique store brand names    : 344
Unique kitchen locations    : 350
Difference (multi-location) : 6


In [17]:
# === Categorical values check ===
categorical_cols = ['CITY', 'STATUS', 'ZONE MAPPING', 'REVENUE COHORT', 
                    'CM COHORT', 'EBITDA CATEGORY', 'EBITDA COHORT']

for col in categorical_cols:
    print(f"\n{col} — {df[col].nunique()} unique values:")
    print(df[col].value_counts())


CITY — 5 unique values:
CITY
Ahmedabad    522
Hyderabad    432
Bangalore    402
Mumbai       378
Pune         366
Name: count, dtype: int64

STATUS — 2 unique values:
STATUS
Active      1116
Inactive     984
Name: count, dtype: int64

ZONE MAPPING — 4 unique values:
ZONE MAPPING
South    582
East     558
West     528
North    432
Name: count, dtype: int64

REVENUE COHORT — 3 unique values:
REVENUE COHORT
More than 40 lacs    733
INR 20 to 30 lacs    690
INR 30 to 40 lacs    677
Name: count, dtype: int64

CM COHORT — 3 unique values:
CM COHORT
More than 30%    771
10% to 20%       708
0% to 10%        621
Name: count, dtype: int64

EBITDA CATEGORY — 2 unique values:
EBITDA CATEGORY
EBITDA -ve    1094
EBITDA +ve    1006
Name: count, dtype: int64

EBITDA COHORT — 3 unique values:
EBITDA COHORT
10% to 20%       707
0% to 10%        703
More than 20%    690
Name: count, dtype: int64


In [18]:
# Check cohort column values
print("Revenue Cohort:")
print(df['REVENUE COHORT'].unique())

print("\nCM Cohort:")
print(df['CM COHORT'].unique())

Revenue Cohort:
<ArrowStringArray>
['INR 20 to 30 lacs', 'INR 30 to 40 lacs', 'More than 40 lacs']
Length: 3, dtype: str

CM Cohort:
<ArrowStringArray>
['0% to 10%', 'More than 30%', '10% to 20%']
Length: 3, dtype: str


In [19]:
print("EBITDA Category:")
print(df['EBITDA CATEGORY'].unique())

print("\nEBITDA Cohort:")
print(df['EBITDA COHORT'].unique())

EBITDA Category:
<ArrowStringArray>
['EBITDA +ve', 'EBITDA -ve']
Length: 2, dtype: str

EBITDA Cohort:
<ArrowStringArray>
['More than 20%', '0% to 10%', '10% to 20%']
Length: 3, dtype: str


### Key Observations

- Ahmedabad has the highest number of records in the dataset (522), while Pune has the lowest (366).
- Active and Inactive stores are relatively balanced (Active: 1116 , Inactive: 984).
- Zone distribution is fairly even across South (582), East (558), West (528), and North (432).
- Revenue Cohort currently contains only 3 categories in the provided data:
  - INR 20 to 30 lacs
  - INR 30 to 40 lacs
  - More than 40 lacs
- CM Cohort does not include a 20%–30% range.
- EBITDA is already categorized into positive and negative groups:
  - EBITDA +ve
  - EBITDA -ve

In [20]:
# === Numeric ranges ===
numeric_cols = ['ORDER COUNT', 'CART SALES', 'DISCOUNT', 'NET REVENUE',
                'IDEAL FOOD COST', 'GROSS MARGIN', 'KITCHEN EBITDA', 'VARIANCE']
df[numeric_cols].describe()

,ORDER COUNT,CART SALES,DISCOUNT,NET REVENUE,IDEAL FOOD COST,GROSS MARGIN,KITCHEN EBITDA,VARIANCE
count,2100.000000,2.100000e+03,2.100000e+03,2.100000e+03,2.100000e+03,2.100000e+03,2.100000e+03,2100.000000
mean,11045.371905,4.400591e+06,9.839992e+05,3.505318e+06,1.362716e+06,2.043380e+06,6.847230e+05,20176.588981
std,2290.859134,1.119388e+06,3.136288e+05,8.948390e+05,4.151767e+05,5.724239e+05,5.810918e+05,5783.204330
min,7002.000000,2.177493e+06,3.521976e+05,1.726080e+06,5.567911e+05,8.628336e+05,-5.858504e+05,10018.480000
25%,9097.000000,3.520108e+06,7.442392e+05,2.817943e+06,1.059971e+06,1.612744e+06,2.520450e+05,15494.855000
50%,10987.000000,4.295395e+06,9.388061e+05,3.427512e+06,1.298579e+06,1.980660e+06,6.176177e+05,20195.190000
75%,12986.750000,5.184657e+06,1.176024e+06,4.127204e+06,1.618120e+06,2.423338e+06,1.076676e+06,25180.827500
max,14999.000000,7.471638e+06,2.147224e+06,6.167425e+06,2.797857e+06,4.172730e+06,2.834680e+06,29993.940000


- All revenue/cost values are positive  — no data corruption
- KITCHEN EBITDA minimum: -₹5.86 lakh/month — confirms loss-making stores exist (aligns with 47% Inactive STATUS)
- NET REVENUE range: ₹17.3 lakh to ₹61.7 lakh per month — the brief's "Below INR 15 lacs" bucket will be empty in this dataset
- VARIANCE: ₹10K-30K (absolute) — must convert to % in Phase 2 to match brief's variance buckets
- ORDER COUNT: 7K-15K per month — reasonable for cloud kitchens (230-500 orders/day)

In [21]:
# === Time Range Analysis ===
df['MONTH_DT'] = pd.to_datetime(
    df['MONTH'],
    format='%b-%Y'
)

print(f"Earliest Month : {df['MONTH_DT'].min().strftime('%b-%Y')}")
print(f"Latest Month   : {df['MONTH_DT'].max().strftime('%b-%Y')}")
print(f"Total Months   : {df['MONTH_DT'].nunique()}")

print("\nRows Per Month:\n")

print(
    df['MONTH_DT']
    .dt.strftime('%b-%Y')
    .value_counts()
    .sort_index()
)

Earliest Month : Oct-2023
Latest Month   : Mar-2024
Total Months   : 6

Rows Per Month:

MONTH_DT
Dec-2023    350
Feb-2024    350
Jan-2024    350
Mar-2024    350
Nov-2023    350
Oct-2023    350
Name: count, dtype: int64


### Phase 2 Preparation

- Convert `MONTH` into datetime format for proper monthly trend analysis.
- Create additional metrics such as GM%, CM%, EBITDA%, and Variance%.
- Recreate Revenue Cohort buckets as per assignment requirements.
- Create variance percentage buckets for variance-level dashboard analysis.
- Use `GROSS MARGIN` as a proxy for Contribution Margin since a separate CM column is not available.
- Use `(CITY, STORE, ZONE)` combination as the operational kitchen identifier for analysis.

In [23]:
# Convert MONTH from string to datetime
df['MONTH'] = pd.to_datetime(df['MONTH'], format='%b-%Y')

# Confirm the conversion
print(f"MONTH dtype : {df['MONTH'].dtype}")
print(f"Min month   : {df['MONTH'].min()}")
print(f"Max month   : {df['MONTH'].max()}")

MONTH dtype : datetime64[us]
Min month   : 2023-10-01 00:00:00
Max month   : 2024-03-01 00:00:00


In [24]:
df = df.drop(columns=['MONTH_DT'])
print(f"Columns after cleanup: {df.shape[1]}")

Columns after cleanup: 17


In [25]:
df['GM %']        = (df['GROSS MARGIN']   / df['NET REVENUE']) * 100
df['CM %']        = (df['GROSS MARGIN']   / df['NET REVENUE']) * 100
df['EBITDA %']    = (df['KITCHEN EBITDA'] / df['NET REVENUE']) * 100
df['VARIANCE %']  = (df['VARIANCE']       / df['NET REVENUE']) * 100

df[['NET REVENUE', 'GM %', 'CM %', 'EBITDA %', 'VARIANCE %']].describe().round(2)

,NET REVENUE,GM %,CM %,EBITDA %,VARIANCE %
count,2100.00,2100.00,2100.00,2100.00,2100.00
mean,3505318.18,58.20,58.20,16.75,0.62
std,894839.02,5.70,5.70,13.08,0.24
min,1726080.29,46.38,46.38,-32.00,0.18
25%,2817943.14,53.38,53.38,8.79,0.43
50%,3427512.13,58.07,58.07,18.10,0.58
75%,4127204.07,62.95,62.95,26.10,0.77
max,6167425.46,69.76,69.76,47.14,1.49


In [27]:
# === Revenue cohort ===
revenue_bins   = [0, 15_00_000, 25_00_000, 35_00_000, 45_00_000, float('inf')]
revenue_labels = ['(a) Below INR 15 lacs',
                  '(b) INR 15 to 25 lacs',
                  '(c) INR 25 to 35 lacs',
                  '(d) INR 35 to 45 lacs',
                  '(e) Above INR 45 lacs']

df['REVENUE CATEGORY'] = pd.cut(df['NET REVENUE'],
                                 bins=revenue_bins,
                                 labels=revenue_labels,
                                 right=False)

In [28]:
# === Variance category ===
variance_bins   = [0, 2, 3, 5, float('inf')]
variance_labels = ['(a) Var < 2%',
                   '(b) Var 2% to 3%',
                   '(c) Var 3% to 5%',
                   '(d) Var > 5%']

df['VARIANCE CATEGORY'] = pd.cut(df['VARIANCE %'],
                                  bins=variance_bins,
                                  labels=variance_labels,
                                  right=False)


In [29]:
# Verify
print("Revenue Category distribution:")
print(df['REVENUE CATEGORY'].value_counts().sort_index())
print("\nVariance Category distribution:")
print(df['VARIANCE CATEGORY'].value_counts().sort_index())

Revenue Category distribution:
REVENUE CATEGORY
(a) Below INR 15 lacs      0
(b) INR 15 to 25 lacs    278
(c) INR 25 to 35 lacs    834
(d) INR 35 to 45 lacs    689
(e) Above INR 45 lacs    299
Name: count, dtype: int64

Variance Category distribution:
VARIANCE CATEGORY
(a) Var < 2%        2100
(b) Var 2% to 3%       0
(c) Var 3% to 5%       0
(d) Var > 5%           0
Name: count, dtype: int64


## Phase 3: Business Insights

We move from understanding the data to drawing conclusions about the business.

### Insight 1 — Do Inactive kitchens lose money?

We saw that 47% of all kitchen-month records are marked Inactive. Are these the loss-making operations, or is the status assignment random?

In [30]:
# Compare Active vs Inactive performance
status_comparison = df.groupby('STATUS').agg(
    avg_revenue   = ('NET REVENUE',    'mean'),
    avg_ebitda    = ('KITCHEN EBITDA', 'mean'),
    avg_ebitda_pct= ('EBITDA %',       'mean'),
    avg_variance  = ('VARIANCE %',     'mean'),
    row_count     = ('STORE',          'count')
).round(2)

status_comparison

,avg_revenue,avg_ebitda,avg_ebitda_pct,avg_variance,row_count
STATUS,,,,,
Active,3500145.93,687547.29,16.94,0.61,1116
Inactive,3511184.27,681519.81,16.53,0.62,984


Finding 1: STATUS is not correlated with profitability.

Active and Inactive kitchens show nearly identical performance:
- Avg revenue differs by ₹11K (0.3%)
- Avg EBITDA% differs by 0.4 percentage points
- Variance % is identical

This contradicts the natural assumption that "Inactive" kitchens were closed due to poor financials. Possible explanations:
- STATUS may reflect operational state (paused, under renovation, franchise transition) rather than financial health
- The field may be stale or not consistently updated
- In a real-world scenario, this would warrant clarification from the business team before using STATUS in any segmentation logic

Implication for dashboard: STATUS should be available as a filter but should not be used as a profitability indicator. Better signals: EBITDA %, EBITDA CATEGORY.

In [31]:
# Compare cities on key profitability metrics
city_performance = df.groupby('CITY').agg(
    avg_revenue    = ('NET REVENUE',    'mean'),
    avg_ebitda     = ('KITCHEN EBITDA', 'mean'),
    avg_ebitda_pct = ('EBITDA %',       'mean'),
    avg_gm_pct     = ('GM %',           'mean'),
    row_count      = ('STORE',          'count')
).round(2).sort_values('avg_ebitda_pct', ascending=False)

city_performance

,avg_revenue,avg_ebitda,avg_ebitda_pct,avg_gm_pct,row_count
CITY,,,,,
Bangalore,3559037.98,699683.75,17.30,57.92,402
Ahmedabad,3540383.38,716593.08,17.26,58.43,522
Hyderabad,3500175.79,690889.77,16.81,58.40,432
Pune,3460682.54,667147.77,16.45,58.38,366
Mumbai,3448859.89,634770.78,15.67,57.76,378


Finding 2: City does not meaningfully drive profitability.

In [32]:
# Aggregate to the kitchen level using unique key
kitchen_performance = df.groupby(['CITY', 'STORE', 'ZONE MAPPING']).agg(
    total_revenue   = ('NET REVENUE',    'sum'),
    total_ebitda    = ('KITCHEN EBITDA', 'sum'),
    avg_ebitda_pct  = ('EBITDA %',       'mean'),
    months_active   = ('MONTH',          'count')
).round(2).reset_index()

# Top 10 by total EBITDA
print("=== TOP 10 KITCHENS by Total EBITDA ===")
kitchen_performance.nlargest(10, 'total_ebitda')

=== TOP 10 KITCHENS by Total EBITDA ===


,CITY,STORE,ZONE MAPPING,total_revenue,total_ebitda,avg_ebitda_pct,months_active
179,Hyderabad,Jimenez LLC,North,29120234.41,9170224.04,31.34,6
75,Ahmedabad,Smith Ltd,West,25471539.62,7961714.87,28.00,6
167,Hyderabad,Fuller LLC,East,27592361.28,7620794.46,25.97,6
48,Ahmedabad,"Mahoney, Dixon and Hobbs",East,26373716.62,7537180.56,26.72,6
0,Ahmedabad,Adams LLC,East,25300755.79,7500022.48,25.97,6
70,Ahmedabad,Rose and Sons,East,25480625.53,7499551.77,28.78,6
232,Mumbai,Barnes PLC,South,25686721.63,7254693.48,24.30,6
347,Pune,"Thompson, Hardy and Ramos",West,25961648.63,7250769.44,26.10,6
209,Hyderabad,Sanders Ltd,North,26712794.16,7180599.28,26.61,6
133,Bangalore,"Price, Blevins and Kim",North,25280039.71,7172797.36,25.33,6


In [33]:
# Bottom 10 by total EBITDA
print("=== BOTTOM 10 KITCHENS by Total EBITDA ===")
kitchen_performance.nsmallest(10, 'total_ebitda')

=== BOTTOM 10 KITCHENS by Total EBITDA ===


,CITY,STORE,ZONE MAPPING,total_revenue,total_ebitda,avg_ebitda_pct,months_active
90,Bangalore,"Andrews, Reed and Silva",South,13421387.86,-779236.85,-6.37,6
294,Pune,Bowman-Martinez,West,17122152.14,517367.27,1.56,6
35,Ahmedabad,"Hansen, Delgado and Bennett",East,16462290.57,658163.51,2.24,6
288,Mumbai,"Wright, Floyd and Taylor",North,16559339.61,854224.27,4.23,6
215,Hyderabad,Thomas LLC,South,17042754.46,979098.09,3.07,6
289,Pune,"Aguilar, Becker and Hernandez",East,18149471.90,1082398.29,2.13,6
277,Mumbai,"Smith, Duffy and White",North,15782509.89,1093576.91,6.41,6
333,Pune,Reyes PLC,North,17380214.93,1131447.30,3.96,6
16,Ahmedabad,Camacho-Hayes,North,17138859.44,1192742.41,2.10,6
84,Ahmedabad,"Wolf, Quinn and Johnson",East,17056738.49,1267902.13,7.03,6


Finding 3: Profitability variance lives at the kitchen level - and 99% of kitchens are profitable.


In [34]:
# Correlation between variance and profitability
correlation_vp = df['VARIANCE %'].corr(df['EBITDA %'])
correlation_vr = df['VARIANCE %'].corr(df['NET REVENUE'])
correlation_vg = df['VARIANCE %'].corr(df['GM %'])

print("=== Correlation of VARIANCE % with key metrics ===")
print(f"VARIANCE % vs EBITDA %    : {correlation_vp:.4f}")
print(f"VARIANCE % vs NET REVENUE : {correlation_vr:.4f}")
print(f"VARIANCE % vs GM %        : {correlation_vg:.4f}")

=== Correlation of VARIANCE % with key metrics ===
VARIANCE % vs EBITDA %    : -0.5880
VARIANCE % vs NET REVENUE : -0.6447
VARIANCE % vs GM %        : -0.0671


Finding 4: Variance (wastage) is a real driver of profitability.

In [35]:
# Saveing data for the Streamlit dashboard
df.to_csv('../data/kitchen_pnl_prepared.csv', index=False)
print(f"Saved: {df.shape[0]} rows × {df.shape[1]} columns")

Saved: 2100 rows × 23 columns
